src/main/java/uk/gov/hmcts/reform/prl/services/ManageOrderEmailService.java

```diff
@@ -459,118 +525,135 @@ private String getCafcassEmail(ManageOrders manageOrders) {
+                                    buildBulkPrintOrderDetail(bulkPrintId, element.getCode(),
+                                            partyData.getLabelForDynamicList()))
+                            );
+                        } else {
+                            log.info("Couldn't send serve order details to respondent, address is null/empty for {}", element.getCode());
+                        }
                     } catch (Exception e) {
-                        log.info("Unable to notify respondent");
-                        throw new RuntimeException(e);
+                        log.error("Error in sending order docs to respondent {}", element.getCode());
+                        log.error("Exception occurred in sending order docs to respondent", e);
                     }
                 }
-
             }
         });
     }
 
-    private void notifyCitizen(SelectTypeOfOrderEnum isFinalOrder, CaseDetails caseDetails, CaseData caseData, Map<String,
-        String> partyMapTemp, Optional<Element<PartyDetails>> partyData, String authorisation) throws Exception {
-        log.info("Inside notifyCitizen while serving order");
-        if (CaseUtils.isCitizenAccessEnabled(partyData.get().getValue())) {
-            log.info("Dashboard access already enabled");
-            if (ContactPreferences.digital.equals(partyData.get().getValue().getContactPreferences())
-                && isPartyProvidedWithEmail(partyData)) {
-                log.info("Contact preference set as email");
-                sendEmailToPartyOrPartySolicitor(
-                    isFinalOrder,
-                    partyMapTemp.entrySet().iterator().next().getKey(),
-                    buildApplicantRespondentEmail(
-                        caseDetails,
-                        partyMapTemp.entrySet().iterator().next().getValue()
-                    ),
-                    caseData
-                );
-            } else {
-                log.info("Contact preference set as post");
-                sendOrderDocumentViaPost(caseData, partyData.get().getValue(), authorisation);
-            }
-        } else {
-            log.info("Dashboard access not enabled");
-            sendOrderDocumentViaPost(caseData, partyData.get().getValue(), authorisation);
-
+    private UUID sendOrderDocumentViaPost(CaseData caseData,
+                                          PartyDetails partyData,
+                                          String authorisation,
+                                          List<Document> orderDocuments) throws Exception {
+        List<Document> documents = new ArrayList<>();
+        //generate cover letter
+        Document coverLetter = serviceOfApplicationPostService.getCoverLetter(
+                caseData,
+                authorisation,
+                partyData.getAddress(),
+                partyData.getLabelForDynamicList()
+        );
+        if (isNotEmpty(coverLetter)) {
```

Comment:
Cover letter should be on top always

Reply:
Hi @shashisk97 yes cover is the first/top document in the list `documents` which is sent to bulk print.
I noticed an NPE in bulk print service when cover letter is not generated so just added null check.